config/v1/types_cluster_version_test.go

```diff
@@ -10,14 +11,49 @@ import (
 func TestKnownClusterVersionCapabilities(t *testing.T) {
 	exists := struct{}{}
 	known := make(map[ClusterVersionCapability]struct{}, len(KnownClusterVersionCapabilities))
-	for _, cap := range KnownClusterVersionCapabilities {
-		known[cap] = exists
+	for _, capability := range KnownClusterVersionCapabilities {
+		known[capability] = exists
 	}
 
 	for set, caps := range ClusterVersionCapabilitySets {
-		for _, cap := range caps {
-			if _, ok := known[cap]; !ok {
-				t.Errorf("Capability set %s contains %s, which needs to be added to KnownClusterVersionCapabilities", set, cap)
+		for _, capability := range caps {
+			if _, ok := known[capability]; !ok {
+				t.Errorf("Capability set %s contains %s, which needs to be added to KnownClusterVersionCapabilities", set, capability)
+			}
+		}
+	}
+}
+
+// TestSubsetsVersionCapabilities verifies that all preceding sets
+// (order given by capabilitySetOrder) are subsets of following sets
+func TestSubsetsVersionCapabilities(t *testing.T) {
+	// defines the order by which subsets are tested
+	var capabilitySetOrder = map[int]ClusterVersionCapabilitySet{
+		0: ClusterVersionCapabilitySetNone,
+		1: ClusterVersionCapabilitySet4_11,
+		2: ClusterVersionCapabilitySetCurrent,
+	}
+
+	var keys []int
+	for k := range capabilitySetOrder {
+		keys = append(keys, k)
+	}
+	sort.Ints(keys)
```

Comment:
/hold

if I were you, I would use a completely different homogeneous data structure.

Reply:
Replaced map for array in [here](https://github.com/openshift/api/pull/1242).

(Moved to new PR due to my mistake in git work)